docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/git-merge-pr.md

```diff
@@ -0,0 +1,73 @@
+---
+sidebar_label: git-merge-pr
+description: Merges a specified open pull request.
+---
+
+# `git-merge-pr`
+
+`git-merge-pr` merges a specified open pull request. This step commonly follows a
+[`git-open-pr` step](git-open-pr.md) or can be used independently to merge existing
+pull requests.
+
+## Configuration
+
+| Name | Type | Required | Description |
+|------|------|----------|-------------|
+| `repoURL` | `string` | Y | The URL of a remote Git repository. |
+| `provider` | `string` | N | The name of the Git provider to use. Currently `azure`, `bitbucket`, `gitea`, `github`, and `gitlab` are supported. Kargo will try to infer the provider if it is not explicitly specified. |
+| `insecureSkipTLSVerify` | `boolean` | N | Indicates whether to bypass TLS certificate verification when interfacing with the Git provider. Setting this to `true` is highly discouraged in production. |
+| `prNumber` | `integer` | Y | The pull request number to merge. |
+| `wait` | `boolean` | N | If `true`, the step will return a running status instead of failing when the PR is not yet mergeable. The merge will be retried on the next reconciliation until it succeeds or times out. Default is `false`. Note: The `wait` option is unreliable for repositories hosted on Bitbucket due to API limitations. |
+
+## Output
+
+| Name | Type | Description |
+|------|------|-------------|
+| `commit` | `string` | The ID (SHA) of the merge commit created after successfully merging the pull request. Typically, a subsequent [`argocd-update` step](argocd-update.md) will reference this output to learn the ID of the commit that an applicable Argo CD `ApplicationSource` should be observably synced to under healthy conditions. |
+
+## Examples
+
+### Basic Usage
+
+In this example, a pull request is merged immediately if it's ready. If the pull
+request is not ready to merge (e.g., due to pending checks or conflicts), the step
+will fail.
+
+```yaml
+steps:
+  - uses: git-merge-pr
+    config:
+      repoURL: https://github.com/example/repo.git
+      prNumber: 42
+```
+
+### Merge with Wait
+
+This example demonstrates merging a pull request with waiting enabled. If the pull
+request is not immediately ready to merge, the step will return a running status and
+Kargo will retry it later.
+
+```yaml
+steps:
+  - uses: git-merge-pr
+    config:
+      repoURL: https://github.com/example/repo.git
+      prNumber: 42
+      wait: true
+```
+
+### Custom Git Provider
+
+This example demonstrates specifying a custom Git provider for `git-merge-pr`. This
+is useful when the provider cannot be inferred from the `repoURL`, such as with
+self-hosted GitLab instances.
```

Comment:
Please figure out how to reword this or even drop this section entirely.

This is _purely_ for instructing Kargo on what kind of provider it's interfacing with when that cannot be inferred. The world "custom," however, strongly implies this is for using providers that aren't otherwise supported, which is not the case.